### PR TITLE
Don't generate stacktrace in EarlyTerminationException and TimeExceededException

### PR DESCRIPTION
**Pull request by iverase**
Those exceptions are only created to terminate a search task and never expose to the user so there is no point to generate the stacktrace.

Inspired by https://github.com/apache/lucene/pull/12270

### CODE REVIEW

**Review by javanna — APPROVED** [review]
left two nits, LGTM

File: server/src/main/java/org/elasticsearch/search/query/EarlyTerminatingCollector.java [hunk 1]
```diff
@@ -21,11 +21,17 @@
 /**
  * A {@link Collector} that early terminates collection after <code>maxCountHits</code> docs have been collected.
  */
-public class EarlyTerminatingCollector extends FilterCollector {
+class EarlyTerminatingCollector extends FilterCollector {
     static final class EarlyTerminationException extends RuntimeException {
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/search/query/EarlyTerminatingCollector.java:25]: this constructor could be private.

File: server/src/main/java/org/elasticsearch/search/query/QueryPhase.java [hunk 2]
```diff
@@ -364,7 +350,34 @@ private static boolean canEarlyTerminate(IndexReader reader, SortAndFormats sort
         return true;
     }
 
-    public static class TimeExceededException extends RuntimeException {}
+    public static Runnable getTimeOutCheck(SearchContext searchContext) {
```
[the remark this answers is not in the document]
Comment (iverase) [reply server/src/main/java/org/elasticsearch/search/query/QueryPhase.java]: oh yes

File: server/src/main/java/org/elasticsearch/search/query/EarlyTerminatingCollector.java (hunk 1 above)
Comment (iverase) [reply server/src/main/java/org/elasticsearch/search/query/EarlyTerminatingCollector.java:25]: done

**Pull request merged**